tests/test_format.py

```diff
@@ -67,6 +67,19 @@ def test_h5():
     }
 
 
+@pytest.mark.h5
+def test_h5_optional_key():
```

Comment:
Sorry, forgot to ask this earlier but could you also make sure that there are some tests for using the builder classes with a default value? I think everything should work without further modifications, but it would be good to add a test to confirm that.

I think it would be good to add a test in `test_builder.py` similar to `test_build_mapped_key_options` but with the default value.

And then it would also be good to update `integration_tests/test_full_example.py` as that is supposed to be basically a giant test covering every possible metadata mapper feature that we have.

Reply:
Added them.